cpp/custom-dataset/README.md

```diff
@@ -0,0 +1,54 @@
+# Custom Dataset Example with the PyTorch C++ Frontend
+
+This folder contains an example of loading a custom image dataset with OpenCV and training a model to label images, using the PyTorch C++ frontend.
+
+The dataset used here is [Caltech 101](http://www.vision.caltech.edu/Image_Datasets/Caltech101/) dataset.
+
+The entire training code is contained in custom-data.cpp.
+
+You can find instructions on how to install OpenCV [here](../tools/InstallingOpenCV.md).
+
+To build the code, run the following commands from your terminal:
+
+```shell
+$ cd custom-dataset
+$ mkdir build
+$ cd build
+$ cmake -DCMAKE_PREFIX_PATH=/path/to/libtorch ..
+$ make
+```
+
+where /path/to/libtorch should be the path to the unzipped LibTorch distribution, which you can get from the [PyTorch homepage](https://pytorch.org/get-started/locally/).
+
```

Comment:
The LibTorch nightly or major release distribution might not be using a compiler that's ABI-compatible with the compiler used for building OpenCV. I think we should add a note here saying that if the user sees an error like `undefined reference to cv::imread(std::string const&, int)` when running the `make` command, they should build LibTorch from source using the instructions at https://github.com/pytorch/pytorch#from-source, and then set `CMAKE_PREFIX_PATH` to that PyTorch source directory.